### PR TITLE
Open main project file in editor when opening project

### PR DESCRIPTION
**Pull request by lslachciak**
[Issue] N/A
[Problem] When opening project, nothing is opened
[Solution] Read mian project file name from config.xml
and open it

Signed-off-by: Lukasz Slachciak <l.slachciak@samsung.com>

### CODE REVIEW

File: libs/brackets-server/embedded-ext/project/main.js [hunk 1]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
         .done(function(data) {
             _typeTable = data;
         });
+    // on opening of every project
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: I think code below is self documented and this comment can be removed.

File: libs/brackets-server/embedded-ext/project/main.js [hunk 2]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+    ProjectManager.on("projectOpen", function () {
+        // if this is project with config.xml
+        var file = FileSystem.getFileForPath(ProjectManager.getProjectRoot().fullPath +"config.xml");
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: var -> const
missing space: +"config.xml"

File: libs/brackets-server/embedded-ext/project/main.js [hunk 3]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+        file.exists(function (err, exists) {
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: Please do console log with error code and early return if err object is true.

File: libs/brackets-server/embedded-ext/project/main.js [hunk 4]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+            if (exists) {
+                FileUtils.readAsText(file)
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: Can we merge those two lines?

File: libs/brackets-server/embedded-ext/project/main.js [hunk 5]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+                .done((text, readTimestamp) => {
+                    // try to extract "content" field describing main project file
+                    let parser = new DOMParser();
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: let -> const

File: libs/brackets-server/embedded-ext/project/main.js [hunk 6]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+                    let xmlDoc = parser.parseFromString(text,"text/xml");
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: Ditto.
Missing space text,"text/xml")

File: libs/brackets-server/embedded-ext/project/main.js [hunk 7]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+                    let elements = xmlDoc.getElementsByTagNameNS("http://www.w3.org/ns/widgets", "content");
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: Ditto.

File: libs/brackets-server/embedded-ext/project/main.js [hunk 8]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+                    if (elements.length > 0) {
+                        let content = elements[0].attributes["src"].nodeValue;
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: Ditto.

File: libs/brackets-server/embedded-ext/project/main.js [hunk 9]
```diff
@@ -32,6 +33,30 @@ define(function (require, exports, module) {
...
+                        if (!!content) {
+                            let htmlPath = ProjectManager.getProjectRoot().fullPath + content;
```
Comment (grzegorz-czajkowski) [inline libs/brackets-server/embedded-ext/project/main.js]: Ditto.

File: libs/brackets-server/embedded-ext/project/main.js (hunk 1 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 2 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 3 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 4 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: one

File: libs/brackets-server/embedded-ext/project/main.js (hunk 5 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 6 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 7 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 8 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

File: libs/brackets-server/embedded-ext/project/main.js (hunk 9 above)
Comment (lslachciak) [reply libs/brackets-server/embedded-ext/project/main.js]: done

**Comment by lslachciak** [comment]
Thank you @grzegorz-czajkowski for review. I updated PR with fixes according to your comments

**Pull request merged**